management/dbaas/xtra_db_cluster_test.go

```diff
@@ -0,0 +1,217 @@
+							CPUm:        2,
+							MemoryBytes: "128",
+						},
+					},
+				},
+			},
+		}
+
+		paramsUpdatePXC.WithContext(pmmapitests.Context)
+		_, err = dbaasClient.Default.XtraDBCluster.UpdateXtraDBCluster(&paramsUpdatePXC)
+		pmmapitests.AssertAPIErrorf(t, err, 501, codes.Unimplemented, `This method is not implemented yet.`)
+
+		for _, pxc := range xtraDBClusters.Payload.Clusters {
+			if pxc.Name == "" {
+				continue
+			}
+			deleteXtraDBClusterParamsParam := xtra_db_cluster.DeleteXtraDBClusterParams{
+				Body: xtra_db_cluster.DeleteXtraDBClusterBody{
+					KubernetesClusterName: kubernetesClusterName,
+					Name:                  pxc.Name,
+				},
+			}
+			deleteXtraDBClusterParamsParam.WithContext(pmmapitests.Context)
+			_, err := dbaasClient.Default.XtraDBCluster.DeleteXtraDBCluster(&deleteXtraDBClusterParamsParam)
+			assert.NoError(t, err)
+		}
+	})
+
+	t.Run("Create_XtraDBCluster_Invalid_Name", func(t *testing.T) {
+		paramsPXCEmptyName := xtra_db_cluster.CreateXtraDBClusterParams{
+			Body: xtra_db_cluster.CreateXtraDBClusterBody{
+				KubernetesClusterName: kubernetesClusterName,
+				Name:                  "",
+				Params: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParams{
+					ClusterSize: 1,
+					Proxysql: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsProxysql{
+						ComputeResources: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsProxysqlComputeResources{
+							CPUm:        1,
+							MemoryBytes: "64",
+						},
+					},
+					Pxc: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsPxc{
+						ComputeResources: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsPxcComputeResources{
+							CPUm:        1,
+							MemoryBytes: "64",
+						},
+					},
+				},
+			},
+		}
+		paramsPXCEmptyName.WithContext(pmmapitests.Context)
+		_, err := dbaasClient.Default.XtraDBCluster.CreateXtraDBCluster(&paramsPXCEmptyName)
+		pmmapitests.AssertAPIErrorf(t, err, 400, codes.InvalidArgument, `invalid field Name: value '' must not be an empty string`)
+
+		paramsPXCInvalidName := xtra_db_cluster.CreateXtraDBClusterParams{
+			Body: xtra_db_cluster.CreateXtraDBClusterBody{
+				KubernetesClusterName: kubernetesClusterName,
+				Name:                  "123_asd",
+				Params: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParams{
+					ClusterSize: 1,
+					Proxysql: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsProxysql{
+						ComputeResources: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsProxysqlComputeResources{
+							CPUm:        1,
+							MemoryBytes: "64",
+						},
+					},
+					Pxc: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsPxc{
+						ComputeResources: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsPxcComputeResources{
+							CPUm:        1,
+							MemoryBytes: "64",
+						},
+					},
+				},
+			},
+		}
+		paramsPXCInvalidName.WithContext(pmmapitests.Context)
+		_, err = dbaasClient.Default.XtraDBCluster.CreateXtraDBCluster(&paramsPXCInvalidName)
+		assert.Error(t, err)
```

Comment:
`pmmapitests.AssertAPIErrorf`?

Reply:
Not everywhere it is possible, as error contains a tmp random path to kubeconfig and depends on kubectl version/path.
I do not see big benefits to make any workaround here right now.

Reply:
ok, can we check for a status code?